include/gsl/gsl

```diff
@@ -117,6 +117,42 @@ std::ostream& operator<<(std::ostream& os, const not_null<T>& val)
     return os;
 }
 
+template <class T, class U>
+bool operator==(const not_null<T>& lhs, const not_null<U>& rhs)
+{
+    return lhs.get() == rhs.get();
+}
```

Comment:
These would be better with return-type expression SFINAE:
```c++
template <class T, class U>
auto operator==(const not_null<T>& t, const not_null<U>& u) -> decltype(t.get() == u.get())
{
    return t.get() == u.get();
}
```
although I suspect VS2013 would explode. @neilmacintosh, is it time yet to drop VS2013 support?